e2e/test/scenarios/metrics/metrics-editing.cy.spec.ts

```diff
@@ -0,0 +1,527 @@
+import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
+import type { StructuredQuestionDetails } from "e2e/support/helpers";
+import {
+  startNewMetric,
+  createQuestion,
+  echartsContainer,
+  enterCustomColumnDetails,
+  getNotebookStep,
+  modal,
+  popover,
+  restore,
+  assertQueryBuilderRowCount,
+} from "e2e/support/helpers";
+
+const { ORDERS_ID, ORDERS, PRODUCTS_ID, PRODUCTS } = SAMPLE_DATABASE;
+
+type QuestionDetails = StructuredQuestionDetails & { name: string };
+
+const ORDER_COUNT_DETAILS: QuestionDetails = {
+  name: "Orders metric",
+  type: "metric",
+  query: {
+    "source-table": ORDERS_ID,
+    aggregation: [["count"]],
+  },
+  display: "scalar",
+};
+
+const ORDER_COUNT_CREATED_AT_DETAILS: QuestionDetails = {
+  name: "Orders metric",
+  type: "metric",
+  query: {
+    "source-table": ORDERS_ID,
+    breakout: [["field", ORDERS.CREATED_AT, { "temporal-unit": "month" }]],
+    aggregation: [["count"]],
+  },
+  display: "scalar",
+};
+
+const PRODUCT_COUNT_DETAILS: QuestionDetails = {
+  name: "Products metric",
+  type: "metric",
+  query: {
+    "source-table": PRODUCTS_ID,
+    aggregation: [["count"]],
+  },
+  display: "scalar",
+};
+
+const PRODUCT_COUNT_CREATED_AT_DETAILS: QuestionDetails = {
+  name: "Products metric",
+  type: "metric",
+  query: {
+    "source-table": PRODUCTS_ID,
+    breakout: [["field", PRODUCTS.CREATED_AT, { "temporal-unit": "month" }]],
+    aggregation: [["count"]],
+  },
+  display: "scalar",
+};
+
+describe("scenarios > metrics", () => {
```

Comment:
`git` isn't smart enough to understand it's the same file. What's changed:
- Helpers got options and they started to work in multiple stages
- Some multi-stage tests were added